chain-impl-mockchain/src/testing/arbitrary/mod.rs

```diff
@@ -64,3 +63,25 @@ impl Arbitrary for Output<Address> {
         }
     }
 }
+
+impl Arbitrary for Milli {
+    fn arbitrary<G: Gen>(g: &mut G) -> Self {
+        Milli::from_millis(u64::arbitrary(g))
+    }
+}
+
+impl Arbitrary for Hash {
+    fn arbitrary<G: Gen>(g: &mut G) -> Self {
+        Hash(Arbitrary::arbitrary(g))
+    }
+}
+
+impl Arbitrary for LinearFee {
+    fn arbitrary<G: Gen>(g: &mut G) -> Self {
+        Self {
+            constant: Arbitrary::arbitrary(g),
+            coefficient: Arbitrary::arbitrary(g),
+            certificate: Arbitrary::arbitrary(g),
+        }
+    }
+}
```

Comment:
I'm not really convinced that moving impls of `Arbitrary` to a central point is a good idea.

1. This file is going to become a behemoth and a messy jungle
2. We're decoupling type definition and instance generation. It's going to be hard to find what type `X`'s `Arbitrary` looks like, because this isn't going to be defined where `X` is.
1. Some `Arbitrary` impls are probably not going to be movable at all, because they can depend on access to private fields.

Reply:
1) it can be split accross testing module. the only requirement was that it needs to be in testing module
2) Yes i have similar doubts. but i couldn't find any other solution for enabling jormungadr integration tests use testing structures in `chain-impl-mockchain`. I copied "solution" for chain-crypto. The thing is, in order to use code from `chain-impl-mockchain` in test cfg in `jormungandr-integration-tests`  i need to import it with :

chain-impl-mockchain = { path = "../chain-addr", features=["property-test-api"]}

and this requires all arbitrary to be visible in testing mod. i couldn;t find any other idea how to fix it.

3) Unfortunately yes, i fixed that by `pub(crate)` but dunno if this is a great idea

Reply:
The 3. alone is a red flag for me. We can't dehermetize things to allow foreign hacky constructors. At most we can provide normal constructors to build hacky ones on top of them.

The trait impl doesn't have to be visible or imported, its existence is enough for it to be used. I think that making them exist by passing a feature isn't a bad idea. The `impl Arbitrary for ...` will have to be annotated (or put into an annotated module) with something like `#[cfg(any(test, feature = "test_utils_please"))]`

Reply:
Oh, great. your solution is working as expected. pushed changes with annotations